wgpu-types/src/lib.rs

```diff
@@ -1303,6 +1358,49 @@ impl<L> TextureDescriptor<L> {
     }
 }
 
+/// Texture descriptor builder methods
+impl<'l> TextureDescriptor<Option<Cow<'l, str>>> {
+    pub fn new(
+        size: Extent3d,
+        dimension: TextureDimension,
+        format: TextureFormat,
+        usage: TextureUsage,
+    ) -> Self {
+        Self {
+            label: None,
+            size,
+            mip_level_count: 1,
+            sample_count: 1,
+            dimension,
+            format,
+            usage,
+        }
+    }
+
+    pub fn build(&self) -> Self {
```

Comment:
is this necessary? If all our methods receive `&Descriptor`, most users don't need this `build()` step.

Reply:
No, not necessarily. I guess people can just use clone if they need to put it somewhere.